doc/sonic-sub-port-intf-hld.md

```diff
@@ -0,0 +1,475 @@
+    }
+},
+```
+A key in the DOT1Q_INTERFACE table is the name of a sub port, which consists of two sections delimited by a "." (symbol dot). The section before the dot is the name of the parent physical port or port channel. The section after the dot is the dot1q encapsulation vlan id.
+
+mtu of a sub port interface should be no greater than that of the parent physical port or port channel. In the case field "mtu" is absent in the config_db.json file, a sub port interface inherits the mtu from its parent physical port or port channel.
+
+Example configuration:
+```
+"DOT1Q_INTERFACE": {
+    "Ethernet64.10": {
+        "mtu" : "9100"
+    },
+    "Ethernet64.10|192.168.0.1/21": {}
+},
+
+"VLAN": {
+    "Vlan10": {
+        "vlanid" : "10"
+    }
+},
+```
+
+### 2.1.2 CONFIG_DB
+```
+DOT1Q_INTERFACE|{{ port_name }}.{{ vlan_id }}
+    "mtu" : "{{ mtu_size }}"
+
+DOT1Q_INTERFACE|{{ port_name }}.{{ vlan_id }}|{{ ip_prefix }}
+    "NULL" : "NULL"
+```
+
+### 2.1.3 CONFIG_DB schemas
+```
+; Defines for sub port interface configuration attributes
+key             = DOT1Q_INTERFACE|subif_name    ; subif_name is the name of the sub port interface
+
+; subif_name annotations
+subif_name      = port_name "." vlan_id         ; port_name is the name of parent physical port or port channel
+                                                ; vlanid is DIGIT 1-4094
+
+; field         = value
+mtu             = 1*4DIGIT                      ; sub port interface MTU (Optional)
+```
+
+```
+; Defines for sub port interface configuration attributes
+key             = DOT1Q_INTERFACE|subif_name|IPprefix   ; an instance of this key will be repeated for each IP prefix
+
+IPprefix        = IPv4prefix / IPv6prefix               ; an instance of this key/value pair will be repeated for each IP prefix
+
+IPv4prefix      = dec-octet "." dec-octet "." dec-octet "." dec-octet "/" %d1-32
+dec-octet       = DIGIT                 ; 0-9
+                 / %x31-39 DIGIT        ; 10-99
+                 / "1" 2DIGIT           ; 100-199
+                 / "2" %x30-34 DIGIT    ; 200-249
+                 / "25" %x30-35         ; 250-255
+
+IPv6prefix      =                             6( h16 ":" ) ls32
+                 /                       "::" 5( h16 ":" ) ls32
+                 / [               h16 ] "::" 4( h16 ":" ) ls32
+                 / [ *1( h16 ":" ) h16 ] "::" 3( h16 ":" ) ls32
+                 / [ *2( h16 ":" ) h16 ] "::" 2( h16 ":" ) ls32
+                 / [ *3( h16 ":" ) h16 ] "::"    h16 ":"   ls32
+                 / [ *4( h16 ":" ) h16 ] "::"              ls32
+                 / [ *5( h16 ":" ) h16 ] "::"              h16
+                 / [ *6( h16 ":" ) h16 ] "::"
+h16             = 1*4HEXDIG
+ls32            = ( h16 ":" h16 ) / IPv4address
+```
+
+Example:
+```
+DOT1Q_INTERFACE|Ethernet64.10
+    "mtu" : "9100"
+
+DOT1Q_INTERFACE|Ethernet64.10|192.168.0.1/21
+    "NULL" : "NULL"
+```
+
+## 2.2 APPL_DB
+```
+INTF_TABLE:{{ port_name }}.{{ vlan_id }}
```

Comment:
Are you reusing an old table? If yes, is it possible that old schema and new schema coexist in one switch?

Reply:
If coexist, is it possible to normalize the schema?

Reply:
It will co-exist. There is no need to normalization as both follows the naming convention.